mars/dataframe/execution/datasource.py

```diff
@@ -14,14 +14,29 @@
 
 from ..expressions.datasource.dataframe import DataFrameDataSource
 from ..expressions.datasource.series import SeriesDataSource
+from ..expressions.datasource.dataframe_from_tensor import DataFrameFromTensor
+
+try:
+    import pandas as pd
+except ImportError:  # pragma: no cover
+    pass
 
 
 def _dataframe_or_series_pandas_data_source(ctx, chunk):
     ctx[chunk.key] = chunk.op.data
 
 
+def _dataframe_tensor_data_source(ctx, chunk):
+    tensor_chunk = ctx[chunk.inputs[0].key]
+    pdf = pd.DataFrame(tensor_chunk)
+    pdf.index = chunk.index_value
+    pdf.columns = chunk.columns
+    ctx[chunk.key] = pdf
```

Comment:
it can be simplified as `pd.DataFrame(data, index=chunk.index_value, columns=chunk.columns)`